### PR TITLE
helm search: new CLI flag `--fail-on-no-result` to fail when no results found for search (repo and hub)

### LINKED ISSUE #7197: Helm search should have the option to exit with 1 if no results found
`helm version`: `version.BuildInfo{Version:"v3.0.1", GitCommit:"7c22ef9ce89e0ebeb7125ba2ebf7d421f3e82ffa", GitTreeState:"clean", GoVersion:"go1.13.4"}`

`helm search repo` and `helm search hub` should have an option to exit with `1` when nothing is found. This is useful in automation scripts. Something like `--fail` or `--fail-if-no-results-found`.

### PR DESCRIPTION
**Pull request by Bhargav-InfraCloud**
**What this PR does / why we need it**:

Add new CLI flag `--fail-on-no-result` for failing the `helm search` when there are no results found. This works with:
1. `helm search repo`
2. `helm search hub`

Closes #7197 

**Special notes for your reviewer**:
- Since this is a new flag altogether, it should be backward-compatible.
- I believe no documentation is required as the flag is self-explanatory.

**If applicable**:
- [ ] this PR contains documentation
- [x] this PR contains unit tests
- [x] this PR has been tested for backwards compatibility

@giggio @hickeyma @roubles @joejulian @gjenkins8 Please validate whether these changes achieve the expected result. Thanks!

### CODE REVIEW

**Review by giggio — APPROVED** [review]

**Review by joejulian — APPROVED** [review]
Tested this and it works as expected. The flag's a bit wordy, but I also like my bike sheds to be hunter green.

**Review by mattfarina — APPROVED** [review]
